File: JavaScript/1002. Find Common Characters.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var commonChars = function(words) {
        var res = new Array();
        for(var c of words[0]){
            var flag = 1;
            for(var i = 1; i < words.length; i++){
                if(words[i].includes(c)){
                    words[i] = words[i].replace(c,""); 
                }
                else{
                    flag = 0;
                    break;
                }
            }
            if(flag!=0){
                res.push(c);
            }
        }
        return res;
};

File: C++/1002. Find Common Characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<string> res;
        for(auto c: words[0]){
            int flag = 1;
            for(int i = 1; i < words.size(); i++){
                if(words[i].find(c)!=string::npos){
                    words[i].erase(words[i].find(c), 1);
                }
                else{
                    flag = 0;
                    break;
                }
            }
            if(flag!=0){
                res.push_back(string(1,c));
            }
        }
        return res;
    }
};
